cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -619,3 +631,82 @@ func getDockerOptionsWithDefaults(options DockerProjectOptions) DockerProjectOpt
 
 	return options
 }
+
+func addDefaultDockerfileForJava(serviceConfig ServiceConfig) (string, error) {
+	// if Dockerfile not exists, provide a default one
+	if serviceConfig.Language == ServiceLanguageJava {
+		log.Printf("Dockerfile not found for java project %s, will provide a default one", serviceConfig.Name)
+		dockerfileDir, err := os.MkdirTemp("", fmt.Sprintf("%s-%d", serviceConfig.Name, time.Now().Unix()))
+		if err != nil {
+			return "", fmt.Errorf("error creating temp Dockerfile directory: %w", err)
+		}
+		hasParent := serviceConfig.ParentPath != ""
+		dockerfilePath, err := writeDockerfileIntoFs(dockerfileDir, hasParent)
+		if err != nil {
+			return "", err
+		}
+		return dockerfilePath, nil
+	}
+
+	return "", fmt.Errorf("adding default Dockerfile is not supported for language: %s", serviceConfig.Language)
+}
+
+// todo: hardcode jdk-21 as base image here, may need more accurate java version detection.
+const (
+	DockerfileSingleStage = `FROM openjdk:21-jdk-slim
+COPY ./target/*.jar app.jar
+COPY ./target/*.war app.war
```

Comment:
How about deleting `.war`, in normal case, `war` file is used to be deployed in web server. And `jar` will be run by `java -jar`.

Reply:
yeah, we can delete it first.